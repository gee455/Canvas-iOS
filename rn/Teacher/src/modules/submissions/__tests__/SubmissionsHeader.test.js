//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow
import React from 'react'
import SubmissionsHeader from '../SubmissionsHeader.js'
import renderer from 'react-test-renderer'
import defaultFilterOptions from '../../filter/filter-options'
import explore from '../../../../test/helpers/explore'

let template = {
  ...require('../../../__templates__/helm'),
}

jest.mock('TouchableOpacity', () => 'TouchableOpacity')

test('SubmissionHeader navigates to filter when pressed', () => {
  let navigator = template.navigator()
  let filterOptions = defaultFilterOptions()
  const tree = renderer.create(
    <SubmissionsHeader filterOptions={filterOptions} navigator={navigator} />
  ).toJSON()

  let filterButton = explore(tree).selectByID('submission-list.filter') || {}
  filterButton.props.onPress()

  expect(navigator.show).toHaveBeenCalledWith(
    '/filter',
    { modal: true },
    { filterOptions, navigator }
  )
})

test('SubmissionHeader anonymous grading', () => {
  const tree = renderer.create(
    <SubmissionsHeader filterOptions={defaultFilterOptions()} anonymous />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('SubmissionHeader muted grading', () => {
  const tree = renderer.create(
    <SubmissionsHeader filterOptions={defaultFilterOptions()} muted />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('SubmissionHeader anonymous and muted', () => {
  const tree = renderer.create(
    <SubmissionsHeader filterOptions={defaultFilterOptions()} anonymous muted />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
