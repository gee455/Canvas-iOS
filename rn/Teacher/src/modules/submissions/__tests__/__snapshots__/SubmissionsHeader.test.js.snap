// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionHeader anonymous and muted 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityTraits="header"
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontFamily": ".SFUIDisplay-semibold",
            "fontSize": 20,
          },
          Object {
            "color": "#2d3b44",
            "fontSize": 20,
            "fontWeight": "600",
          },
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
          Object {
            "fontFamily": ".SFUIDisplay-medium",
          },
        ]
      }
    >
      Grades muted, Anonymous grading
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions"
    accessibilityTraits="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {},
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SubmissionHeader anonymous grading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityTraits="header"
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontFamily": ".SFUIDisplay-semibold",
            "fontSize": 20,
          },
          Object {
            "color": "#2d3b44",
            "fontSize": 20,
            "fontWeight": "600",
          },
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
          Object {
            "fontFamily": ".SFUIDisplay-medium",
          },
        ]
      }
    >
      Anonymous grading
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions"
    accessibilityTraits="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {},
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SubmissionHeader muted grading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityTraits="header"
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontFamily": ".SFUIDisplay-semibold",
            "fontSize": 20,
          },
          Object {
            "color": "#2d3b44",
            "fontSize": 20,
            "fontWeight": "600",
          },
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
          Object {
            "fontFamily": ".SFUIDisplay-medium",
          },
        ]
      }
    >
      Grades muted
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions"
    accessibilityTraits="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {},
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
